bigquery_etl/backfill/shredder_mitigation.py

```diff
@@ -0,0 +1,193 @@
+"""Generate a query to backfill an aggregate with shredder mitigation."""
+
+from datetime import date, datetime, time, timedelta
+from enum import Enum
+from types import NoneType
+
+TEMP_DATASET = "tmp"
+SUFFIX = datetime.now().strftime("%Y%m%d%H%M%S")
+PREVIOUS_DATE = (datetime.now() - timedelta(days=1)).strftime("%Y-%m-%d")
+
+
+class ColumnType(Enum):
+    """Differentiate metric and dimensions."""
+
+    METRIC = "METRIC"
+    DIMENSION = "DIMENSION"
+    UNDETERMINED = "None"
+
+
+class ColumnStatus(Enum):
+    """Different status of a column during shredder mitigation."""
+
+    COMMON = "COMMON"
+    ADDED = "ADDED"
+    REMOVED = "REMOVED"
+    UNDETERMINED = "None"
+
+
+class DataTypeGroup(Enum):
+    """Data types in BigQuery. Not including ARRAY and STRUCT as these are not expected in aggregate tables."""
+
+    STRING = ("STRING", "BYTES")
+    BOOLEAN = "BOOLEAN"
+    NUMERIC = (
+        "INTEGER",
+        "NUMERIC",
+        "BIGNUMERIC",
+        "DECIMAL",
+        "INT64",
+        "INT",
+        "SMALLINT",
+        "BIGINT",
+        "TINYINT",
+        "BYTEINT",
+    )
+    FLOAT = ("FLOAT",)
```

Comment:
The separation of FLOAT from other numeric values is to facilitate rounding columns with this dtype in order to avoid exponential numbers when computing very small numbers.